store/sqlstore/post_persistent_notification_store.go

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/store"
+	"github.com/mattermost/mattermost-server/v6/utils"
+	sq "github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+)
+
+type SqlPostPersistentNotificationStore struct {
+	*SqlStore
+}
+
+func newSqlPostPersistentNotificationStore(sqlStore *SqlStore) store.PostPersistentNotificationStore {
+	return &SqlPostPersistentNotificationStore{
+		SqlStore: sqlStore,
+	}
+}
+
+func (s *SqlPostPersistentNotificationStore) Get(params model.GetPersistentNotificationsPostsParams) ([]*model.PostPersistentNotifications, bool, error) {
+	andCond := sq.And{
+		sq.Eq{"DeleteAt": 0},
+	}
+
+	if params.PostID != "" {
+		andCond = append(andCond, sq.Eq{"PostId": params.PostID})
+	}
+	if params.MaxCreateAt > 0 {
+		andCond = append(andCond, sq.LtOrEq{"CreateAt": params.MaxCreateAt})
+	}
+	if params.MaxLastSentAt > 0 {
+		andCond = append(andCond, sq.LtOrEq{"LastSentAt": params.MaxLastSentAt})
+	}
+
+	builder := s.getQueryBuilder().
+		Select("*").
+		From("PersistentNotifications").
+		Where(andCond)
+
+	builder = getPersistentNotificationsPaginationBuilder(builder, params.Pagination)
+
+	var posts []*model.PostPersistentNotifications
+	err := s.GetReplicaX().SelectBuilder(&posts, builder)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return []*model.PostPersistentNotifications{}, false, nil
+		}
+		return nil, false, errors.Wrap(err, "failed to get notifications")
+	}
+
+	var hasNext bool
+	if params.Pagination.PerPage != 0 {
+		if len(posts) == utils.MinInt(params.Pagination.PerPage, paginationLimit)+1 {
+			// Shave off the next-page item.
+			posts = posts[:len(posts)-1]
+			hasNext = true
+		}
+	}
+	return posts, hasNext, nil
+}
+
+// Delete in batches of 1000
```

Comment:
Since, these methods are available via APIs, although indirectly, can be used by users to call them with any big N no. to crash the server. But I guess even for that they might need to create these N channels/teams in the first place.

So, I'll remove the batches. 🙂